src/Features/Core/Portable/ExtractInterface/AbstractExtractInterfaceService.cs

```diff
@@ -298,29 +391,46 @@ private static Solution GetSolutionWithFormattedInterfaceDocument(Document unfor
             // extracted members are also not updated.
             if (typeToExtractFrom.TypeKind == TypeKind.Interface)
             {
-                return solutionWithFormattedInterfaceDocument;
+                return solution;
             }
 
-            var formattedSolution = GetSolutionWithUpdatedOriginalType(
-                solutionWithFormattedInterfaceDocument,
+            var unformattedSolution = solution;
+            foreach (var documentId in documentIds)
+            {
+                var document = solution.GetDocument(documentId);
+                var currentRoot = document.GetSyntaxRootSynchronously(cancellationToken);
+                var editor = new SyntaxEditor(currentRoot, solution.Workspace);
```

Comment:
editor has editor.GetGenerator

Reply:
Does it? I wasn't able to find it with http://source.roslyn.io/#q=GetSyntaxGenerator or http://source.roslyn.io/#q=GetGenerator